woocommerce/woocommerce-ios
# [Composite Products] Load component options data in Component Settings view

Part of #8956


## Description
This PR loads the **list of component options** and **default option** in the Component Settings view, for Composite Products. Component options can be products or categories, while the default option is always a specific product. This means that products always need to be loaded (at least to get the default option), and categories sometimes need to be loaded.

When the data is loaded:

* Products are synced from remote (in case not all products have been synced yet).
* Categories are fetched from storage since they are fully synced on the product details screen.

Note: We will add loading error handling in another PR.

### Changes

* Loads component options using the option IDs and option type (products or categories).
* Fetches the name of the default option from remote (this is always a product).
* Shows a loading view in the component options and default option sections while the loading is in progress.

## Testing instructions

### Prerequisites

1. Install and activate the [Composite Products extension](https://woocommerce.com/products/composite-products/) on your store.
2. Create at least one product with the Composite Products type, with at least one component.

### To test

1. Build and run the app.
2. Go to the Products tab.
3. Open a Composite Product.
4. Select the Components row in product details.
5. Select a component from the list.
6. When the Component Settings view opens, confirm it shows loading indicators for the component options and default option, which are replaced by the actual data when loading is complete.

## Screenshots


https://user-images.githubusercontent.com/8658164/229860817-d65968b5-3af3-43fa-85ce-4d0373fc8bc6.mp4



---
